### DISCUSSION TITLE
How can I set up code hints for libraries like SFML?

### DISCUSSION THREAD

**Discussion by Sillybum** [helix-editor/helix#12296] [Q&A] [answered]
Hello, guys!
I love how Helix works but I'm stuck with a problem for hours.
Helix doesn't give any hints when it comes to external libraries like SFML.
I can compile and run the code so the code is working.
I've done everything I could find from googling but couldn't figure out how.
Could anyone please help me?

**Comment by TornaxO7** [comment]
In this case, you have to see what the LSP does. Take a look into the logs while working with helix (you can get the path to the log from the help page: `hx --help`) and increase the verbosity level by adding `-v` during your execution (for example: `hx -vvv`).

**Reply by Sillybum** [reply]
That's a great information, thank you!

**Comment by lpkha21** [comment, marked as answer]
you can use bear or cmake to generate "compile_commands.json", which the editor will read and suggest the functions.
in cmake this is the command: set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
to generate that file.

**Reply by Sillybum** [reply]
Thank you for your answer! I generated the json file and the code completion is working well 😭
